samples/ControlCatalog.NetCore/Program.cs

```diff
@@ -53,22 +53,31 @@ static int Main(string[] args)
             else if (args.Contains("--full-headless"))
             {
                 return builder
-                    .UseHeadless(true)
+                    .UseHeadless(new AvaloniaHeadlessPlatformOptions
+                    {
+                        UseHeadlessDrawing = true,
+                        UseCompositor = true
+                    })
                     .AfterSetup(_ =>
                     {
+                        static Task LowPriorityDelay(int ms)
+                        {
+                            return Task.Delay(ms).ContinueWith(_ =>
+                                    Dispatcher.UIThread.InvokeAsync(() => { }, DispatcherPriority.Background))
+                                .Unwrap();
+                        }
                         DispatcherTimer.RunOnce(async () =>
                         {
                             var window = ((IClassicDesktopStyleApplicationLifetime)Application.Current.ApplicationLifetime)
                                 .MainWindow;
                             var tc = window.GetLogicalDescendants().OfType<TabControl>().First();
                             foreach (var page in tc.Items.Cast<TabItem>().ToList())
                             {
-                                // Skip DatePicker because of some layout bug in grid
-                                if (page.Header.ToString() == "DatePicker")
+                                if (page.Header.ToString() == "DatePicker" || page.Header.ToString() == "TreeView")
                                     continue;
                                 Console.WriteLine("Selecting " + page.Header);
                                 tc.SelectedItem = page;
-                                await Task.Delay(500);
+                                await LowPriorityDelay(20);
```

Comment:
I understand this is a workaround for an unrelated bug. We should remove it and fix the bug ;)